File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        unordered_map<int,int>mp; 
        vector<int>ans; 

        for( auto num : nums){
            if( mp[num] < 2) ans.push_back(num); 
            mp[num]++;
        }
        nums = ans; 
        return ans.size(); 
        
    }
};

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
import java.util.*;

class Solution {
    public int removeDuplicates(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        List<Integer> ans = new ArrayList<>();

        for (int num : nums) {
            int count = mp.getOrDefault(num, 0);
            if (count < 2) {
                ans.add(num);
                mp.put(num, count + 1);
            }
        }

        // Copy modified list back to original array
        for (int i = 0; i < ans.size(); i++) {
            nums[i] = ans.get(i);
        }

        return ans.size(); // New length of array
    }
}
